### ISSUE TITLE
Sync preview to current cursor doesn't work for included files

### ISSUE THREAD

**Issue by 7sDream** [Enter-tainer/typst-preview#164]
**Describe the bug**

In a included typ file, the sync preview with current cursor command is not working

**To Reproduce**

`main.typ` file:
```typ
main file content

#include "other.typ"
```

`other.typ` file:
```typ
other content
```

1. Start the preview from `main.typ` file.
2. Open `other.type`, make sure it's the actived editor.
3. Run "Sync preview with current cursor" command, nothing happened.

**Expected behavior**

Preview scroll to cursor pos of other.typ file.

**Package/Software version:**

VSCode version(Help -> About):
```plain
Version: 1.83.1 (user setup)
Commit: f1b07bd25dfad64b0167beb15359ae573aecd2cc
Date: 2023-10-10T23:48:05.904Z
Electron: 25.8.4
ElectronBuildId: 24154031
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.25982
```

**Additional context**

I briefly studied the current implementation code and found that the reason is that the VSCode extension code maintains an “activeTask”, which is a “doc --> server” mapping. When executing the preview sync command, it checks whether the current document is in this mapping. If it is not, do nothing.

As a temporary solution for personal use, I tried a [small modification](https://github.com/Enter-tainer/typst-preview/compare/v0.8.3...7sDream:typst-preview:patch/include-file-sync-cursor): When there is only one item in activeTask, the preview sync event in any file will be sent to the one unique preview server. This works fine for me and fully meets my needs, I usually only open one preview server.

If we need support multiple preview instances and achieve it theoretically correct(considering troublesome situations such as multiple preview instances depending on the same file), the extension code needs dependency information from compiler. The implementation may require modifications to the upstream projects, like typst.ts, etc. So I just opened this Issue for discussion instead of write a PR.

**Comment by Myriad-Dreamin** [comment]
For quick solution, we can simply broadcast request to all preview server. This could introduce false positive (FP), that some other preview panels accidentally scroll to position. But it will exactly work on most scenario that:
+ there is only single preview panel, hence no FP.
+ the request is on a content file only owned by the some preview panel, hence no FP.

The false positive is indeed introduced when we make request on a template or package file shared by multiple preview panels, which are less likely to happen. We can wait for users that need this further improvement. @Enter-tainer

**Comment by Enter-tainer** [comment]
Both the broadcast and https://github.com/Enter-tainer/typst-preview/compare/v0.8.3...7sDream:typst-preview:patch/include-file-sync-cursor looks good to me.

To fully address this issue, the vscode extension code should be aware of the files used by `typst-preview` server, which is not easy.

**Comment by 7sDream** [comment]
Okay, I'll happy to send a PR of my changes first so that I can use official release instead of my fork.

For the final solution, we can keep this Issue open and discuss it in the long term.

**Comment by Enter-tainer** [comment]
closed as #166 merged
